src/components/Root/FXHR.js

```diff
@@ -16,12 +17,24 @@ export default (deps) => {
     }
 
     send = async (payload) => {
+      const { tokenExpiration, logger } = this.FFetchContext;
       this.FFetchContext.logger?.log('rtr', 'capture XHR send');
       if (this.shouldEnsureToken) {
-        await rtr(this.FFetchContext)
-          .then(() => {
-            super.send(payload);
-          });
+        if (!isValidAT(tokenExpiration, logger)) {
+          logger.log('rtr', 'local tokens expired; fetching from storage for XHR..');
+          this.FFetchContext.tokenExpiration = await getTokenExpiry();
+        }
+
+        if (isValidRT(this.FFetchContext.tokenExpiration, logger)) {
+          logger.log('rtr', 'local refresh token valid, sending XHR...');
+          super.send(payload);
+        } else {
+          await rtr(this.FFetchContext)
+            .then(() => {
+              logger.log('rtr', 'local token refreshed, sending XHR...');
+              super.send(payload);
+            });
+        }
```

Comment:
This doesn't seem right to me. If I'm reading correctly, it only does rtr when it thinks the RT is expired. That might work because we shrink the TTL when we store it to avoid rotating tokens right at the edge of their validity, but still... I think we want logic similar to [FFetch::ffetch](https://github.com/folio-org/stripes-core/blob/238b09157f2f0b085bce8891080e3c0cde127353/src/components/Root/FFetch.js#L233-270):
```
if (localAT is expired)
  refetch from storage in case somebody else updated it
if (localAT is valid)
  super.send()
else if (localRT is valid)
  await rtr()
  super.send()
else 
  all is lost; dispatch an RTR_ERROR_EVENT
```